libs/libc/netdb/lib_dnsinit.c

```diff
@@ -140,56 +137,24 @@ bool dns_initialize(void)
  * Name: dns_semtake
  *
  * Description:
- *   Take the DNS semaphore, ignoring errors due to the receipt of signals.
+ *   Take the DNS lock, ignoring errors due to the receipt of signals.
  *
  ****************************************************************************/
 
 void dns_semtake(void)
```

Comment:
Can be a `#define dns_semtake() nxrmutex_lock(&g_dns_lock)`

Reply:
g_dns_lock is gloal